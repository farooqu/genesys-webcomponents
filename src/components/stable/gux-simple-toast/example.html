<div>
  <h3>Accents</h3>
  <gux-simple-toast accent="neutral">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>

  <gux-simple-toast accent="positive">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">
      This is the message that is versy long but is on one line
    </div>
  </gux-simple-toast>

  <gux-simple-toast accent="alert">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">
      <div>This is the first line</div>
      <div>This is the second line</div>
    </div>
  </gux-simple-toast>
  <gux-simple-toast accent="warning">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>
</div>

<div class="stop-default">
  <h3>Dismiss event default prevented</h3>
  <gux-simple-toast accent="neutral">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>
  <gux-simple-toast accent="neutral">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>
  <gux-simple-toast accent="neutral">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>
  <gux-simple-toast accent="neutral">
    <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
    <div slot="message">This is the message</div>
  </gux-simple-toast>
</div>

<style
  onload="(function() {
  const test = document.querySelector('.stop-default');
  test.addEventListener('guxdismiss', (event) => {
    event.preventDefault();

    event.target.classList.toggle('red');
  })
})()"
>
  .red {
    background-color: red !important;
  }
</style>
