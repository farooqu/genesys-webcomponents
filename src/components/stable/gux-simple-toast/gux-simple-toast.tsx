import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  Host,
  JSX,
  Prop
} from '@stencil/core';

import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
import { trackComponent } from '../../../usage-tracking';

import modalComponentResources from './i18n/en.json';
import { GuxSimpleToastAccent } from './gux-simple-toast.types';

/**
 * @slot icon - Required slot for gux-icon
 * @slot message - Required slot for the simple toast message
 */
@Component({
  styleUrl: 'gux-simple-toast.less',
  tag: 'gux-simple-toast'
})
export class GuxSimpleToast {
  /**
   * The component accent.
   */
  @Prop()
  accent: GuxSimpleToastAccent = 'neutral';

  @Event()
  guxdismiss: EventEmitter<void>;

  private getI18nValue: GetI18nValue;

  @Element()
  private root: HTMLElement;

  async componentWillLoad(): Promise<void> {
    trackComponent(this.root, { variant: this.accent });
    this.getI18nValue = await buildI18nForComponent(
      this.root,
      modalComponentResources
    );
  }

  render(): JSX.Element {
    return (
      <Host>
        <div class={`gux-icon gux-${this.accent}`}>
          <slot name="icon" />
        </div>

        <div class="gux-message">
          <slot name="message" />
        </div>

        <button
          class="gux-dismiss-button"
          title={this.getI18nValue('dismiss')}
          onClick={this.onDismissClickHandler.bind(this)}
        >
          <gux-icon
            screenreader-text={this.getI18nValue('dismiss')}
            icon-name="close"
          ></gux-icon>
        </button>
      </Host>
    );
  }

  private onDismissClickHandler(event: MouseEvent): void {
    event.stopPropagation();

    const dismissEvent = this.guxdismiss.emit();
    if (!dismissEvent.defaultPrevented) {
      this.root.remove();
    }
  }
}
