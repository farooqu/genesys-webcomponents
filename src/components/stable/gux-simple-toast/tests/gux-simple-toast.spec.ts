import { newSpecPage } from '@stencil/core/testing';
import { GuxSimpleToast } from '../gux-simple-toast';

const components = [GuxSimpleToast];
const language = 'en';

describe('gux-simple-toast', () => {
  describe('#render', () => {
    [
      {
        description: 'should render neutral simple toast',
        html: `
          <gux-simple-toast accent="neutral">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="message">This is the message</div>
          </gux-simple-toast>
        `
      },
      {
        description: 'should render positive simple toast',
        html: `
          <gux-simple-toast accent="positive">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="message">This is the message</div>
          </gux-simple-toast>
        `
      },
      {
        description: 'should render alert simple toast',
        html: `
          <gux-simple-toast accent="alert">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="message">This is the message</div>
          </gux-simple-toast>
        `
      },
      {
        description: 'should render warning simple toast',
        html: `
          <gux-simple-toast accent="warning">
            <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
            <div slot="message">This is the message</div>
          </gux-simple-toast>
        `
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.rootInstance).toBeInstanceOf(GuxSimpleToast);

        expect(page.root).toMatchSnapshot();
      });
    });
  });

  describe('dismiss', () => {
    it('click dismiss button', async () => {
      const html = `
        <gux-simple-toast accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="message">This is the message</div>
        </gux-simple-toast>
      `;
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const dismissButton =
        element.shadowRoot.querySelector('gux-dismiss-button');
      const guxdismissSpy = jest.fn();
      const clickSpy = jest.fn();
      const elementRemoveSpy = jest.spyOn(element, 'remove');

      page.win.addEventListener('guxdismiss', guxdismissSpy);
      page.win.addEventListener('click', clickSpy);

      dismissButton.click();
      await page.waitForChanges();

      expect(guxdismissSpy).toHaveBeenCalled();
      expect(clickSpy).not.toHaveBeenCalled();
      expect(elementRemoveSpy).toBeCalledWith();
    });

    it('click dismiss button and prevent default', async () => {
      const html = `
        <gux-simple-toast accent="neutral">
          <gux-icon slot="icon" icon-name="user-add" decorative></gux-icon>
          <div slot="message">This is the message</div>
        </gux-simple-toast>
      `;
      const page = await newSpecPage({ components, html, language });
      const element = page.root as HTMLElement;
      const dismissButton =
        element.shadowRoot.querySelector('gux-dismiss-button');
      const elementRemoveSpy = jest.spyOn(element, 'remove');

      page.win.addEventListener('guxdismiss', (event: Event) => {
        event.preventDefault();
      });

      dismissButton.click();
      await page.waitForChanges();

      expect(elementRemoveSpy).not.toBeCalled();
    });
  });
});
