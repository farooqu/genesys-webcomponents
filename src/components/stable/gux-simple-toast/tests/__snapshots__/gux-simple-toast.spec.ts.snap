// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-simple-toast #render should render alert simple toast 1`] = `
<gux-simple-toast accent="alert">
  <!---->
  <div class="gux-alert gux-icon">
    <gux-icon decorative="" icon-name="user-add" slot="icon"></gux-icon>
  </div>
  <div class="gux-message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="gux-dismiss-button" title="Dismiss">
    <gux-icon icon-name="close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render neutral simple toast 1`] = `
<gux-simple-toast accent="neutral">
  <!---->
  <div class="gux-icon gux-neutral">
    <gux-icon decorative="" icon-name="user-add" slot="icon"></gux-icon>
  </div>
  <div class="gux-message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="gux-dismiss-button" title="Dismiss">
    <gux-icon icon-name="close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render positive simple toast 1`] = `
<gux-simple-toast accent="positive">
  <!---->
  <div class="gux-icon gux-positive">
    <gux-icon decorative="" icon-name="user-add" slot="icon"></gux-icon>
  </div>
  <div class="gux-message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="gux-dismiss-button" title="Dismiss">
    <gux-icon icon-name="close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;

exports[`gux-simple-toast #render should render warning simple toast 1`] = `
<gux-simple-toast accent="warning">
  <!---->
  <div class="gux-icon gux-warning">
    <gux-icon decorative="" icon-name="user-add" slot="icon"></gux-icon>
  </div>
  <div class="gux-message">
    <div slot="message">
      This is the message
    </div>
  </div>
  <button class="gux-dismiss-button" title="Dismiss">
    <gux-icon icon-name="close" screenreader-text="Dismiss"></gux-icon>
  </button>
</gux-simple-toast>
`;
